traffic/TrafficCounter.py

```diff
@@ -0,0 +1,133 @@
+import paho.mqtt.publish as publish
+import grovepi
+import json
+from time import sleep
+from configparser import ConfigParser
+
+
+# NOTE:
+# The sensitivity can be adjusted by the onboard potentiometer
+# Calibration
+#       Hold the sensor facing an empty dark area
+#       Adjust the potentiometer until the onboard LED switches off
+#       When the LED switches off the sensor is calibrated
+#       If not, adjust the potentiometer again
+# IMPORTANT: When assigning the port to the sensor assign the next port
+#           from where the sensor has been connected
+#           e.g.: if sensor connectet to d4 assign the sensor to port d5
+#           otherweise the sensor won't work. also leave port d5 and d3 free
+#           to ensure the sensor works correctly
+
+# Connect the Grove Infrared Distance Interrupt Sensor (ir) to digital port D5
+# (see above note)
+
+# read values from config ini
+file = 'config.ini'
+config = ConfigParser()
+config.read(file)
+
+
+# topic constructor
+def topic_constructor(protocol, service_api_key, sensor_id):
+    return ("/" + protocol + "/" + service_api_key + "/" + sensor_id)
+
+
+def attr_topic(topic):
+    return (topic + "/attrs")
+
+
+def cmd_topic(topic):
+    return (topic + "/cmd")
+
+
+def ack_topic(cmd_topic):
+    return (cmd_topic + "exe")
+
+
+# config connection
+broker_address = str(config['connection']['broker_address'])
+port = int(config['connection']['port'])
+username = str(config['connection']['username'])
+password = str(config['connection']['password'])
+# authentication object needed to publish on topic
+auth = {'username': username, 'password': password}
+# topic base config
+protocol = str(config['connection']['protocol'])
+service_api_key = str(config['Traffic_counter']['traffic_service_api_key'])
+
+# VCNL 4010 IR Sensor setup
+# Connect the ir sensor at port d6 NOT d7 (see initial note)
+ir_pin = int(config['Traffic_counter']['ir_pin'])
+grovepi.pinMode(ir_pin, "INPUT")
+ir_id = str(config['Traffic_counter']['ir_id'])
+ir_topic = topic_constructor(protocol, service_api_key, ir_id)
+
+
+# Obstacle sensor set up
+# connect the obstacle sensor to port d8
+os_pin = int(config['Traffic_counter']['os_pin'])
+grovepi.pinMode(os_pin, "INPUT")
+os_id = str(config['Traffic_counter']['os_id'])
```

Comment:
let's be consistent Traffic_counter -> traffic_counter

Reply:
done